iGoByBus/iGoByBus/GBBDataRequestService.m

```
@@ -0,0 +1,145 @@
+    
+    NSDictionary* headers = [configurations[@"configurationHeaders"] objectAtIndex:0];
+    
+    for (NSString* keys in [headers allKeys]) {
+        [self.requestSerializer setValue:headers[keys]
+                      forHTTPHeaderField:keys];
+    }
+}
+
+-(void)startUrlValues:(NSDictionary*)configurations{
+    self.urlFindRoutesByStopName = configurations[@"findByStopName"];
+    self.urlFindStopsByRouteId = configurations[@"findByRouteId"];
+    self.urlDeparturesByRouteId = configurations[@"findDeparturesByRouteId"];
+}
+
+- (void)getParamsFromRequestParameter:(NSString *)requestParameter url_p:(NSString**)url_p parameters_p:(NSDictionary **)parameters_p type:(RequestType)type {
+    switch (type) {
+        case ByRouteId:
+            *parameters_p = [self getParametersForSearchByRouteId:requestParameter];
+            *url_p = self.urlFindStopsByRouteId;
+            break;
+        case ByStopName:
+            *parameters_p = [self getParametersForSearchByStopName:requestParameter];
+            *url_p = self.urlFindRoutesByStopName;
+            break;
+        case DepartureByRouteId:
+            *parameters_p = [self getParametersForSearchDeparturesByRouteId:requestParameter];
+            *url_p = self.urlDeparturesByRouteId;
+            break;
+        default:
+            break;
+    }
+}
+
+-(void)doRequestWithParameter:(NSString*)requestParameter withRequestType:(RequestType)type{
+    NSDictionary *parameters;
+    NSString *url;
+    [self getParamsFromRequestParameter:requestParameter url_p:&url parameters_p:&parameters type:type];
+    
+    [self POST:url parameters:parameters success:^(NSURLSessionDataTask *task, id responseObject) {
+        
+        [self.dataDelegate finishedRequest:responseObject withType:type];
+        
+    } failure:^(NSURLSessionDataTask *task, NSError *error) {
+        
+        [self.dataDelegate requestFailed];
```

You are swallowing the error here.

https://en.wikipedia.org/wiki/Error_hiding